src/Cli.Tests/EnvironmentTests.cs

```diff
@@ -0,0 +1,175 @@
+        Assert.IsNotNull(resolvedJson);
+        Assert.IsTrue(JToken.DeepEquals(
+            JObject.Parse(@"{""envValue"": ""DEVELOPMENT""}"),
+            JObject.Parse(resolvedJson)), "JSON resolved with environment variable correctly");
+    }
+
+    /// <summary>
+    /// This test setups a environment variable in the system and also creates a .env file containing
+    /// same variable with different value to show the value stored in .env file is given
+    /// precedence over the system variable.
+    /// </summary>
+    [TestMethod]
+    public void TestPrecedenceOfEnvironmentFileOverExistingVariables()
+    {
+        // The variable set in the .env file takes precedence over the environment value set in the system.
+        Environment.SetEnvironmentVariable(TEST_ENV_VARIABLE, "TEST");
+
+        // Creating environment variable file
+        File.Create(".env").Close();
+        File.WriteAllText(".env", $"{TEST_ENV_VARIABLE}=DEVELOPMENT");
+        DotNetEnv.Env.Load();     // It contains value DEVELOPMENT
+        Assert.AreEqual("DEVELOPMENT", Environment.GetEnvironmentVariable(TEST_ENV_VARIABLE));
+
+        // If a variable is not present in the .env file then the system defined variable would be used if defined.
+        Environment.SetEnvironmentVariable("HOSTING_TEST_ENVIRONMENT", "PHOENIX_TEST");
+        string? resolvedJson = RuntimeConfigPath.ParseConfigJsonAndReplaceEnvVariables(
+            @"{
+                ""envValue"": ""@env('DAB_TEST_ENVIRONMENT')"",
+                ""hostingEnvValue"": ""@env('HOSTING_TEST_ENVIRONMENT')""
+                }"
+        );
+        Assert.IsNotNull(resolvedJson);
+        Assert.IsTrue(JToken.DeepEquals(
+            JObject.Parse(
+                @"{
+                    ""envValue"": ""DEVELOPMENT"",
+                    ""hostingEnvValue"": ""PHOENIX_TEST""
+                }"),
+            JObject.Parse(resolvedJson)), "JSON resolved with environment variable correctly");
+
+        // Removing the .env file it will then use the value of system environment variable.
+
+    }
+
+    /// <summary>
+    /// Test to verify that no error is thrown if .env file is not present, and existing system variables is used.
+    /// </summary>
+    [TestMethod]
+    public void TestSystemEnvironmentVariableIsUsedInAbsenceOfEnvironmentFile()
+    {
+        Environment.SetEnvironmentVariable(TEST_ENV_VARIABLE, "TEST");
+        Assert.IsFalse(File.Exists(".env"));
+        DotNetEnv.Env.Load(); // No error is thrown
+        Assert.AreEqual("TEST", Environment.GetEnvironmentVariable(TEST_ENV_VARIABLE));
+    }
+
+    /// <summary>
+    /// Test to verify that if the environment variables are not resolved correctly, runtime engine will not start.
+    /// Here, in the first scenario, engine failed to start because the variable defined in the environment file
+    /// is typed incorrectly and do not match with the one present in the config.
```

Comment:
```suggestion
    /// Here, in the first scenario, engine fails to start because the variable defined in the environment file
    /// is typed incorrectly and does not match the one present in the config.
```
nit